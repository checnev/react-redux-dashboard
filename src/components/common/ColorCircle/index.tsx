import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

interface ColorCircleProps {
  color: string;
  width?: string | number;
  height?: string | number;
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  root: {
    display: 'inline-block',
    borderRadius: 100,
  }
}));

const ColorCircle: React.FC<ColorCircleProps> = ({
  color,
  width = 10,
  height = 10
}: ColorCircleProps) => {
  const classes = useStyles();
  return (
    <div className={classes.root} style={{
      width,
      height,
      backgroundColor: color,
    }} />
  );
}; 

export default ColorCircle;
