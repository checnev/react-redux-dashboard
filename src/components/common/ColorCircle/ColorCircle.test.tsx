import { render, unmountComponentAtNode } from 'react-dom';
import { act }  from 'react-dom/test-utils';
import pretty from 'pretty';
import ColorCircle from 'components/common/ColorCircle';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('ColorCiclre render', () => {
  act(() => {
    render(<ColorCircle width={15} height={15} color="#000"/>, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
