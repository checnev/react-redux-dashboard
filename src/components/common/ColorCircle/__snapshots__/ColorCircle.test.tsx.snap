// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorCiclre render 1`] = `"<div class=\\"makeStyles-root-1\\" style=\\"width: 15px; height: 15px; background-color: rgb(0, 0, 0);\\"></div>"`;
